File: Project_Sourcecode_go/output/strategy/makeWithdraw/MakeWithdraw1.go
package makeWithdraw

import (
	"cs586_project/datastore"
	"fmt"
)

type MakeWithdraw1 struct {
}

func (s *MakeWithdraw1) MakeWithdraw(ds interface{}) {
	ds1 := ds.(*datastore.DS1)
	b := ds1.Balance
	w := ds1.Temp_w
	if b-w <= 100 {
		ds1.Balance = b - w - 15
	} else {
		ds1.Balance = b - w
	}
	fmt.Printf("Make Withdraw: %v\n", w)
}

File: Project_Sourcecode_java/src/main/java/edu/iit/cs586/project/output/strategy/MakeWithdraw/MakeWithdraw1.java
package edu.iit.cs586.project.output.strategy.MakeWithdraw;

import edu.iit.cs586.project.datastore.DS1;
import edu.iit.cs586.project.datastore.DataStore;

/**
 * @author CharmingLee <charminglee@jieztech.com>
 * @date 2022/5/2 23:24
 */
public class MakeWithdraw1 implements MakeWithdraw{
    @Override
    public void makeWithdraw(DataStore ds) {
        int b = ((DS1)ds).b;
        int w = ((DS1)ds).temp_w;
        if (b - w <= 100) {
            ((DS1)ds).b = b - w - 15;
        } else {
            ((DS1)ds).b = b - w;
        }
    }
}
